analyses/ssd/flood_trigger/ssd_floodscan_rubkonafangak.md

```diff
@@ -0,0 +1,493 @@
+)
+) +
+stat_smooth(
+geom = "area",
+span = 1/4,
+fill = "#ef6666"
+) +
+scale_x_date(
+date_breaks = "3 months",
+date_labels = "%b"
+) +
+facet_wrap(
+as.formula(paste("~", facet_col)),
+scales="free_x",
+ncol=5
+) +
+ylab("Flooded fraction")+
+xlab("Month")+
+labs(title = title)+
+theme_minimal()
+}
+```
+
+```python
+iso3="ssd"
+config=Config()
+parameters = config.parameters(iso3)
+country_data_raw_dir = Path(config.DATA_DIR) / config.PUBLIC_DIR / config.RAW_DIR / iso3
+country_data_exploration_dir = Path(config.DATA_DIR) / config.PRIVATE_DIR / "exploration" / iso3
+country_data_processed_dir = Path(config.DATA_DIR) / config.PUBLIC_DIR / "processed" / iso3
+country_data_public_exploration_dir = Path(config.DATA_DIR) / config.PUBLIC_DIR / "exploration" / iso3
+bentiu_bound_path=country_data_processed_dir / "bentiu" / "bentiu_bounding_box.gpkg"
+adm2_bound_path=country_data_raw_dir / config.SHAPEFILE_DIR / parameters["path_admin2_shp"]
+```
+
+```python
+gdf_adm2=gpd.read_file(adm2_bound_path)
+```
+
+```python
+fs_clip=xr.load_dataset(country_data_exploration_dir/'floodscan'/f'{iso3}_floodscan.nc')
+#I dont fully understand why, these grid mappings re-occur and what they mean
+#but if having them, later on getting crs problems when computing stats
+fs_clip.SFED_AREA.attrs.pop('grid_mapping')
+fs_clip.NDT_SFED_AREA.attrs.pop('grid_mapping')
+fs_clip.LWMASK_AREA.attrs.pop('grid_mapping')
+fs_clip=fs_clip.rio.write_crs("EPSG:4326",inplace=True)
+```
+
+```python
+da_clip=fs_clip.SFED_AREA
+```
+
+### Stats on Bentiu IDP Camp
+We first compute the statistics for Bentiu only. Note that this is a very small area where there are only 3 cells touching the region and none with the centre in the region.  
+
+```python
+gdf_bentiu=gpd.read_file(bentiu_bound_path)
+```
+
+```python
+#check how many cells are included in the region
+#NOTE: no cells have their centre within the area.. Only 3 touching it
+da_clip.rio.clip(gdf_bentiu.geometry, all_touched = True)
+```
+
+```python
+df_floodscan_bent=compute_raster_statistics(
+        gdf=gdf_bentiu,
+        bound_col="id",
+        raster_array=da_clip,
+        lon_coord="lon",
+        lat_coord="lat",
+        stats_list=["median","min","mean","max","sum","count"],
+        #computes value where 20% of the area is above that value
+        percentile_list=[80],
+        all_touched=True,
+    )
+df_floodscan_bent['year']=df_floodscan_bent.time.dt.year
+df_floodscan_bent['month'] = pd.DatetimeIndex(df_floodscan_bent['time']).month
+df_floodscan_bent['mean_rolling']=df_floodscan_bent.sort_values('time').mean_id.rolling(10,min_periods=10).mean()
+```
+
+We can plot the data over all years. 
+We see that it is pretty rare for Bentiu to get flooded. However, in 2007, 2008, and 2021 a large part of Bentiu did see flooding.  
+Also noteabily the water in 2008 and 2022 didn't recede during the beginning of the year. 
+
+```python
+#note: the 2022 data x-axis is not correct but too hard to fix for purpose
```

Comment:
Yeah but because of smoothing the plot, it doesn't really work. I bet there must be a solution for it but after trying it for a bit without succes I will stop. @caldwellst do you know a trick to do this (this can wait till after your mission!)

Reply:
Hey hey! There's actually no automatic way to do it when using `stat_smooth()` to my knowledge. It's likely given the # of observations that since we aren't specifying the smoothing that it's `stats::loess()`, so the best option would be to create the dataset to infill until the end of 2022 and infill the data using `loess()` with `span = 1/4`. To get the best curve, I think infilling with the latest observation out to the end of 2022 is probably best (can easily use `tidyr::fill()`), and just make those values `NA` **after** smoothing. Then we can just pass this smoothed vector to `geom_area()` in the same manner as we've done before, but we won't see what I assume is a big drop in the area at the end of our observed values.